agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1782,7 +1787,18 @@ bool backhaul_manager::handle_1905_discovery_query(ieee1905_1::CmduMessageRx &cm
     }
 
     std::get<1>(supportedServiceTuple) =
-        wfa_map::tlvSupportedService::eSupportedService::MULTI_AP_CONTROLLER;
+        wfa_map::tlvSupportedService::eSupportedService::MULTI_AP_AGENT;
+
+    if (local_master) {
+        auto supportedServiceTuple = tlvSupportedService->supported_service_list(1);
+        if (!std::get<0>(supportedServiceTuple)) {
+            LOG(ERROR) << "Failed accessing supported_service_list";
+            return false;
+        }
+
+        std::get<1>(supportedServiceTuple) =
+            wfa_map::tlvSupportedService::eSupportedService::MULTI_AP_CONTROLLER;
```

Comment:
Should be `wfa_map::tlvSupportedService::eSupportedService::MULTI_AP_CONTROLLER | wfa_map::tlvSupportedService::eSupportedService::MULTI_AP_AGENT` since in local GW we are also running local agent.

Reply:
I think there must be one item in the list for each supported role, it is not a bitmask.
The case that is not supported is that in which the only supported role is 'controller'

Reply:
Of course, you're right. So in the case we are local_gw we add another supported role which is controller?

Reply:
I don't know how to handle the case where the only supported role is controller. If I've understood it correctly, backhaul_manager_thread is created by the agent process so agent role is always in the list and, if local_master is true, then controller role is also added. Controller code should, if not running together with an agent, also respond to the topology request adding the controller role to the list of the supported roles.